### PR TITLE
`make container-build` support GOPROXY env

### PR DESCRIPTION
**Pull request by FOWind**
Related issue: #295 

1. Modify makefile to support GOPROXY on `container-build` command
2. Add `GOPROXY` use case in README

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,7 +29,7 @@ GO_IMAGE ?= golang:1.16
 default: $(BIN)
 
 container-build:
-	docker run --rm -e VERSION_OVERRIDE=${VERSION_OVERRIDE} -v $(PWD):$(PACKAGE_GOPATH) -w $(PACKAGE_GOPATH) $(GIT_STORAGE_MOUNT) ${GO_IMAGE} /bin/bash -c "make"
+	docker run --rm -e VERSION_OVERRIDE=${VERSION_OVERRIDE}  -e GOPROXY=$(GOPROXY) -v $(PWD):$(PACKAGE_GOPATH) -w $(PACKAGE_GOPATH) $(GIT_STORAGE_MOUNT) ${GO_IMAGE} /bin/bash -c "make"
```
Comment (rifelpet) [inline Makefile:32]: I believe this should work:

```suggestion
	docker run --rm -e VERSION_OVERRIDE=${VERSION_OVERRIDE}  -e GOPROXY -v $(PWD):$(PACKAGE_GOPATH) -w $(PACKAGE_GOPATH) $(GIT_STORAGE_MOUNT) ${GO_IMAGE} /bin/bash -c "make"
```

**Review by rifelpet — APPROVED** [review]
/lgtm

**Comment by abctaylor** [comment]
Just some thoughts: the front page `README` references goproxy.io. Some further digging shows it's sponsored by an entity called `yunzhanghu.com`.

There are k8s estates out there with very stringent infosec requirements. When I was reading the build instructions, I stopped with the project after being told to use goproxy.io and investigating it a bit further. I have no particular view on `yunzhanghu.com` which may be a CDN goproxy.io uses, and I'd be as cynical for any third party service telling me to proxy all my Go dependencies via a third party.

I get why it exists (for users who are struggling due to "Great Firewall" issues, but this may be limiting adoption of what might be a good project.

**Comment by FOWind** [comment]
> Just some thoughts: the front page `README` references goproxy.io. Some further digging shows it's sponsored by an entity called `yunzhanghu.com`.
> 
> There are k8s estates out there with very stringent infosec requirements. When I was reading the build instructions, I stopped with the project after being told to use goproxy.io and investigating it a bit further. I have no particular view on `yunzhanghu.com` which may be a CDN goproxy.io uses, and I'd be as cynical for any third party service telling me to proxy all my Go dependencies via a third party.
> 
> I get why it exists (for users who are struggling due to "Great Firewall" issues, but this may be limiting adoption of what might be a good project.

I think you're right.  You can update the readme file to remove the particular goproxy  uri by a PR.

**Pull request merged**